src/app/(tabs)/search/index.tsx

```diff
@@ -15,46 +15,56 @@ import { SafeAreaView } from 'react-native-safe-area-context';
 
 import styles from './styles';
 import FilterModal from '../../../components/FilterModal/FilterModal';
+import LandingCard from '../../../components/LandingCard/LandingCard';
 import SearchCard from '../../../components/PreviewCard/PreviewCard';
 import RecentSearchCard from '../../../components/RecentSearchCard/RecentSearchCard';
+import { fetchGenres } from '../../../queries/genres';
 import { fetchAllStoryPreviews } from '../../../queries/stories';
-import { StoryPreview, RecentSearch } from '../../../queries/types';
+import { StoryPreview, RecentSearch, Genre } from '../../../queries/types';
 import globalStyles from '../../../styles/globalStyles';
 
-const getRecentSearch = async () => {
-  try {
-    const jsonValue = await AsyncStorage.getItem('GWN_RECENT_SEARCHES_ARRAY');
-    return jsonValue != null ? JSON.parse(jsonValue) : null;
-  } catch (error) {
-    console.log(error);
-  }
-};
-
-const setRecentSearch = async (searchResult: RecentSearch[]) => {
-  try {
-    const jsonValue = JSON.stringify(searchResult);
-    await AsyncStorage.setItem('GWN_RECENT_SEARCHES_ARRAY', jsonValue);
-  } catch (error) {
-    console.log(error);
-  }
-};
-
 function SearchScreen() {
   const [allStories, setAllStories] = useState<StoryPreview[]>([]);
+  const [allGenres, setAllGenres] = useState<Genre[]>([]);
   const [searchResults, setSearchResults] = useState<StoryPreview[]>([]);
   const [search, setSearch] = useState('');
   const [filterVisible, setFilterVisible] = useState(false);
   const [recentSearches, setRecentSearches] = useState<RecentSearch[]>([]);
```

Comment:
The behavior of the search screen should be: genre carousels shown when first navigating to the search screen. When the search bar is pressed/focused, we should see the recents view (genre carousels should be hidden). As a user types, search cards are shown. When the search bar is empty, recents should be shown. When cancel is pressed, recents are hidden and the genre carousels are shown.  

We're going to need to add a useState to help better manage all of the things we have going on on the search screen. Like this:
`const [showRecents, setShowRecents] = useState(false);`

Add this attribute to the SearchBar so that the appropriate change happens when the user clicks on the search bar:
```
            onFocus={() => {
              setShowRecents(true);
              setShowGenreCarousals(false);
            }}
```

I've included the other things that should be added to support this functionality in other comments.